### PR TITLE
Feat/project documents

### PR DESCRIPTION
**Pull request by AlexZorkin**

### CODE REVIEW

File: schema/deploy/tables/attachment.sql [hunk 1]
```diff
@@ -1,12 +1,19 @@
...
 create table cif.attachment
 (
   id integer primary key generated always as identity,
   file uuid,
-  file_name varchar(1000)
+  description varchar(10000),
+  file_name varchar(1000),
+  file_type varchar(100),
+  file_size varchar(100),
+  cif_user_id integer not null references cif.cif_user(id),
```
Comment (matthieu-foucault) [inline schema/deploy/tables/attachment.sql:14]: the `upsert_timestamp_columns` below inserts a `created_by` column, so this is not needed

File: app/pages/cif/project/[project]/attachments.tsx [hunk 2]
```diff
@@ -1,31 +1,60 @@
...
+// ];
+
+const tableFilters = [
+  new NoHeaderFilter(),
+  new TextFilter("File Name", "fileName"),
+  new TextFilter("Type", "type"),
+  new TextFilter("Size", "size"),
+  new TextFilter("Uploaded by", "uploadedBy"),
+  new TextFilter("Status Added", "statusAdded"),
+  new TextFilter("Flag for Review", "temp"),
```
Comment (matthieu-foucault) [inline app/pages/cif/project/[project]/attachments.tsx:54]: Let's not add this "Flag for Review" column for now

File: app/components/Attachment/AttachmentTableRow.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+  return (
+    <tr>
+      <td>{fileName}</td>
+      <td>{fileType}</td>
+      <td>{fileSize}</td>
+      <td>{fullName}</td>
+      <td>{createdAt}</td>
+      <td>
+        <Button onClick={handleViewClick}>View</Button>
```
Comment (matthieu-foucault) [inline app/components/Attachment/AttachmentTableRow.tsx:56]: This should be a `Link`

File: app/lib/helpers/bytesToText.tsx [hunk 4]
```diff
@@ -0,0 +1,11 @@
+export default function formatBytes(bytes, decimals = 2) {
+  if (bytes === 0) return "0 Bytes";
```
Comment (matthieu-foucault) [inline app/lib/helpers/bytesToText.tsx:2]:
```suggestion
  if (bytes <= 0) return "0 Bytes";
```

File: app/lib/helpers/bytesToText.tsx [hunk 5]
```diff
@@ -0,0 +1,11 @@
+export default function formatBytes(bytes, decimals = 2) {
```
Comment (matthieu-foucault) [inline app/lib/helpers/bytesToText.tsx:1]:
```suggestion
export default function formatBytes(bytes: number, decimals = 2) {
```

File: app/components/Attachment/AttachmentTableRow.tsx [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+        <Button onClick={handleDownloadClick} disabled>
+          Download
+        </Button>
```
Comment (matthieu-foucault) [inline app/components/Attachment/AttachmentTableRow.tsx:59]: why is this a button instead of a link to the file to download?

Comment (pbastia) [reply app/components/Attachment/AttachmentTableRow.tsx:59]: I disabled the button, it's not functional. the whole download piece needs some rework

File: app/server/index.ts [hunk 7]
```diff
@@ -53,6 +54,24 @@ app.prepare().then(async () => {
 
   server.use(graphQlMiddleware());
 
+  server.get("/download/:uuid", async (req, res) => {
```
[the remark this answers is not in the document]
Comment (pbastia) [reply app/server/index.ts:57]: I'll leave this out for the download PR

**Review by matthieu-foucault — DISMISSED** [review]
Let's get this merged and improve on it in subsequent PRs

**Review by matthieu-foucault — APPROVED** [review]

**Pull request merged**